soc/xtensa/intel_adsp/common/soc.c

```diff
@@ -281,13 +275,17 @@ static int soc_init(const struct device *dev)
 	sys_write32(LPGPDMA_CHOSEL_FLAG | LPGPDMA_CTLOSEL_FLAG,
 		    DSP_INIT_LPGPDMA(1));
 #endif
+}
 
-	soc_set_power_and_clock();
+static int soc_init(const struct device *dev)
+{
+	if (IS_ENABLED(CONFIG_SOC_SERIES_INTEL_CAVS_V15)) {
+		power_init_v15();
+	} else {
+		power_init();
+	}
 
-#if CONFIG_MP_NUM_CPUS > 1
 	soc_idc_init();
```

Comment:
Because the hardware is present either way and still needs to be initialized.  If you look at the implementation of soc_idc_init() it's just enumerating over CONFIG_MP_NUM_CPUS anyway, so there's minimal wasted effort.